doc/VERSION.md

```diff
@@ -59,24 +96,34 @@ a complex topic, so better don't underestimate it, but
 generally said the methods should behave on the same data
 the same way.
 
+As we use [symbol versioning](dev/symbol-versioning.md), the SO_VERSION
+of Elektra always remains the same, even if the major version changes.
+That means, if we rename or remove a function in the `2.0` release,
+applications that linked against Elektra `1.0` can still link against
+Elektra `2.0` but they do not necessarily compile with Elektra `2.0`
```

Comment:
This makes it seem as if we intend to maintain any function we add to the API indefinitely. We should define a limit, for how many versions removed symbols will be supported. e.g. If we remove `x` in 2.0, it is available for dynamic linking until 3.0 is released, but then it is removed entirely.

Of course there can be exceptions, where it makes sense to keep things around longer, but I think we should limit our guarantees.

Also AFAIK "linking against" in this context only includes the runtime part of dynamic linking. Taking an already compiled `.o` file and only calling `ld` won't work AFAIK. I may be wrong, but we should clarify, what exactly "linking against" means.